From ee0d377ad45891a78dda9915085e14a6207f2464 Mon Sep 17 00:00:00 2001
From: pmooney <Phil_Mooney@cable.comcast.com>
Date: Wed, 15 Nov 2017 22:24:00 +0000
Subject: [PATCH] RDKB-14763,DELIA-24496, DELIA-24645 : remove md4 from openssl
 and wpa-supplicant

From: 3f2444ef92ac72b96271482fa8df013a54ffa970 Dec 18, 2017 7:56 PM

	remove unapproved ciphers from openssl

Source: COMCAST

Upstream-Status: Pending

Signed-off-by: pmooney <Phil_Mooney@cable.comcast.com>

---
 src/crypto/crypto_openssl.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/crypto/crypto_openssl.c b/src/crypto/crypto_openssl.c
index 82c8576..24ecd52 100644
--- a/src/crypto/crypto_openssl.c
+++ b/src/crypto/crypto_openssl.c
@@ -247,7 +247,11 @@ static int openssl_digest_vector(const EVP_MD *type, size_t num_elem,
 int md4_vector(size_t num_elem, const u8 *addr[], const size_t *len, u8 *mac)
 {
 	openssl_load_legacy_provider();
-	return openssl_digest_vector(EVP_md4(), num_elem, addr, len, mac);
+#ifndef OPENSSL_NO_MD4
+        return openssl_digest_vector(EVP_md4(), num_elem, addr, len, mac);
+#else
+       return -1;
+#endif
 }
 #endif /* CONFIG_FIPS */
 
